File: ios/Classes/AndroidMinimizerPlugin.swift
/*
 *  SwiftRadioPlayerPlugin.swift
 *
 *  Created by Ilia Chirkunov <xc@yar.net> on 27.11.2023.
 */

import Flutter
import UIKit

public class AndroidMinimizerPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "android_minimizer", binaryMessenger: registrar.messenger())
    let instance = AndroidMinimizerPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "minimize":
      result(1)
      break
    default:
      result(FlutterMethodNotImplemented)
    }  
  }
}

File: android/src/main/kotlin/com/cheebeez/android_minimizer/AndroidMinimizerPlugin.kt
/*
 *  BackButtonBehaviorPlugin.kt
 *
 *  Created by Ilya Chirkunov <xc@yar.net> on 27.11.2023.
 */

package com.cheebeez.android_minimizer

import androidx.annotation.NonNull
import android.app.Activity;
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import io.flutter.embedding.engine.plugins.activity.ActivityAware;
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;

/** AndroidMinimizerPlugin */
class AndroidMinimizerPlugin: FlutterPlugin, MethodCallHandler, ActivityAware {
  private lateinit var channel : MethodChannel
    private lateinit var activity: Activity;

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "android_minimizer")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(call: MethodCall, result: Result) {
    if (call.method == "minimize") {
      activity.moveTaskToBack(true)
      result.success(1)
    } else {
      result.notImplemented()
    }
  }

  override fun onAttachedToActivity(binding: ActivityPluginBinding) {
    this.activity = binding.activity
  }

  override fun onDetachedFromActivity() {}

  override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
    this.activity = binding.activity
  }

  override fun onDetachedFromActivityForConfigChanges() {}

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}
